ebpf/disk-failure/injection.bpf.c

```diff
@@ -0,0 +1,119 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+// +build ignore
+#include "injection.bpf.h"
```

Comment:
I'm wondering why we have both this header and the includes below. Isn't this doing the same includes?

Reply:
Where did we get these headers. Did you write them @aymericDD ? (that's crazy, they're so big)

Reply:
I will clean and check if I can reduce the size of the header but for my point of view, all this includes are required. I have started my project from this project I have fork and adapted for our needs: https://github.com/lizrice/libbpfgo-beginners
For the `vmlinux.h` it is generate with this command line to reduce the number of include. For more information: https://nakryiko.com/posts/bpf-portability-and-co-re/#btf

Reply:
Is it better now? 😇  @clairecng @ptnapoleon

Reply:
I'm still not understanding why do we use `injection.bpf.h` here? What does the header file do, we have almost the same imports in both file?

Reply:
I have moved all includes into the `injection.bpf.h`, but it's completely arbitrary. If it's better to have all includes into the`.c` file I can remove the `.h`.

Reply:
So I must have had an error with github and didn't see the includes have been moved 😅. Sorry! It's all good

Reply:
To make it clearer we could even make a `include/` directory

Reply:
Agree 💯 We can create a generic header file `includes/ebpf.h` which contains all includes to do an eBPF programs. 👍  I will rework this with the new architecture in the futures PR updates.

Reply:
> I will clean and check if I can reduce the size of the header but for my point of view, all this includes are required. I have started my project from this project I have fork and adapted for our needs: https://github.com/lizrice/libbpfgo-beginners
For the vmlinux.h it is generate with this command line to reduce the number of include. For more information: https://nakryiko.com/posts/bpf-portability-and-co-re/#btf

Has this info made it into the docs? Especially how to re-generate these to pull in newer changes?

Reply:
I am working on a new architecture to manage multiple eBPF projects in order to mutualise common files and facilitate the creation of new eBPF programs. I will create a documentation to explain how to maintain this kind of projects and the `vmlinux.h` is part of this new documentation. 😄

Reply:
I have added two new documentation:
- [How to create an eBPF program and how to maintain it?](https://github.com/DataDog/chaos-controller/pull/638/files#diff-cdc03d89262f04c0bf9712e585e272f316f04056fdd9457c08332b109eef5132)
- [How the disk failure works?](https://github.com/DataDog/chaos-controller/pull/638/files#diff-4612c4cd259ad6c74da7059ea59967c11235f7866226c91c3cd29d853418f681 )